openmls/src/extensions/external_sender_extension.rs

```diff
@@ -21,15 +23,48 @@ pub struct ExternalSender {
     credential: Credential,
 }
 
+impl ExternalSender {
+    /// Creates a new `ExternalSender` instance.
+    pub fn new(signature_key: SignaturePublicKey, credential: Credential) -> Self {
+        Self {
+            signature_key,
+            credential,
+        }
+    }
+
+    pub(crate) fn credential(&self) -> &Credential {
+        &self.credential
+    }
+
+    pub(crate) fn signature_key(&self) -> &SignaturePublicKey {
+        &self.signature_key
+    }
+}
+
 /// ExternalSender (extension data)
 ///
 /// ```c
 /// // draft-ietf-mls-protocol-16
 /// ExternalSender external_senders<V>;
 /// ```
-// TODO(884): Remove `#[allow(unused)]` when #884 is closed.
-#[allow(unused)]
 pub type ExternalSendersExtension = Vec<ExternalSender>;
+/// Identifies an external sender in the `ExternalSendersExtension`.
+#[derive(
+    Debug, PartialEq, Eq, Copy, Clone, Serialize, Deserialize, TlsSerialize, TlsDeserialize, TlsSize,
+)]
+pub struct SenderExtensionIndex(u32);
+
+impl SenderExtensionIndex {
+    /// Creates a new `SenderExtensionIndex` instance.
+    pub fn new(index: u32) -> Self {
+        Self(index)
+    }
+
+    /// Returns the internal index as usize
+    pub(crate) fn index(&self) -> Result<usize, std::num::TryFromIntError> {
```

Comment:
We can cast here since we [enforce a 32bit minimum](https://github.com/openmls/openmls/blob/main/openmls/src/lib.rs#L143) for the platform. That way we avoid the `Result`.

Reply:
Okay. I've changed it to casting